### ISSUE TITLE
Set returning function (SRF) throws error if a single column is used

### ISSUE THREAD

**Issue by JelteF** [tcdi/pgrx#1091]
The following code from the SRF example is modified to remove all but one column from the result set:
```rust
#[pg_extern]
fn calculate_human_years() -> Result<
    TableIterator<'static, (name!(dog_name, Result<Option<String>, pgx::spi::Error>),)>,
    spi::Error,
> {
    let query = "SELECT * FROM dog_daycare;";

    Spi::connect(|client| {
        let mut results = Vec::new();
        let mut tup_table = client.select(query, None, None)?;

        while let Some(row) = tup_table.next() {
            let dog_name = row["dog_name"].value::<String>();
            results.push((dog_name,));
        }

        Ok(TableIterator::new(results.into_iter()))
    })
}
```

And it then throws this error when executed:
```sql
> select * from calculate_human_years();
ERROR:  XX000: return type must be a row type
DETAIL:
LOCATION:  pgx::srf::<impl pgx::iter::TableIterator<_>>::srf_next::{{closure}}, srf.rs:90
```

**Comment by ccleve** [comment]
See #1031

**Comment by eeeebbbbrrrr** [comment]
This was fixed in PR #1164
